server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocator.java

```diff
@@ -190,7 +212,24 @@ protected void reconcile(DesiredBalance desiredBalance, RoutingAllocation alloca
             logger.debug("Reconciling desired balance for [{}]", desiredBalance.lastConvergedIndex());
         }
         allocationOrdering.retainNodes(getNodeIds(allocation.routingNodes()));
-        new DesiredBalanceReconciler(desiredBalance, allocation, allocationOrdering).run();
+        recordTime(cumulativeReconciliationTime, new DesiredBalanceReconciler(desiredBalance, allocation, allocationOrdering)::run);
+    }
+
+    public DesiredBalance getDesiredBalance() {
```

Comment:
`getDesiredBalance` doesn't seem to used everywhere. Is it intentional to expose it?

Reply:
It is not used yet. It is exposed so that it could be used by the API